### DISCUSSION TITLE
LLVM Error - first try out

### DISCUSSION THREAD

**Discussion by jantma2** [mitsuba-renderer/mitsuba3#804] [Q&A] [answered]
Hello,

I try to get started on Mitsuba and want to run a render with LLVM the first time on my windows machine.
I installed the latest win64 LLVM package and let the system write the PATH variable, as suggested.
But I still get this message

```
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
File [~\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\mitsuba\__init__.py:107](https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/Martin/Desktop/~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0/LocalCache/local-packages/Python311/site-packages/mitsuba/__init__.py:107), in MitsubaVariantModule.__getattribute__(self, key)
    104 with _dr.scoped_rtld_deepbind():
    105     modules = (
    106         _import('mitsuba.mitsuba_ext'),
--> 107         _import('mitsuba.mitsuba_' + variant + '_ext'),
    108     )
    110 super().__setattr__('_modules', modules)

File [C:\Program](file:///C:/Program) Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.1264.0_x64__qbz5n2kfra8p0\Lib\importlib\__init__.py:126, in import_module(name, package)
    125         level += 1
--> 126 return _bootstrap._gcd_import(name[level:], package, level)

File :1204, in _gcd_import(name, package, level)

File :1176, in _find_and_load(name, import_)

File :1147, in _find_and_load_unlocked(name, import_)

File :676, in _load_unlocked(spec)

File :573, in module_from_spec(spec)

File :1233, in create_module(self, spec)
...
    117     # Remove those modules from _sys.modules as only the
    118     # MitsubaVariantModule instance should hold a reference to them.
    119     _sys.modules.pop('mitsuba.mitsuba_ext', None)

AttributeError: jit_init_thread_state(): the LLVM backend is inactive because the LLVM shared library ("LLVM-C.dll") could not be found! Set the DRJIT_LIBLLVM_PATH environment variable to specify its path.
Output is truncated. View as a [scrollable element](command:cellOutput.enableScrolling?283577ce-80d4-4dcb-83cb-ef01af87099d) or open in a [text editor](command:workbench.action.openLargeOutput?283577ce-80d4-4dcb-83cb-ef01af87099d). Adjust cell output [settings](command:workbench.action.openSettings?%5B%22%40tag%3AnotebookOutputLayout%22%5D)...
```

I tried several ways of setting the DRJIT_LIBLLVM_PATH variable manually but nothing works. Now, three days after first occurance I lost all hope to solve this on my own. Does anyone know how to proceed?
Thanks a lot.

BR

Edit: By the way this is my code:
```python
import mitsuba as mi
import matplotlib.pyplot as plt
#mi.set_variant('scalar_spectral')
mi.set_variant('llvm_ad_rgb')
#scene = mi.load_file('scenes/living-room-2/scene.xml')
#image = mi.render(scene)
img = mi.render(mi.load_dict(mi.cornell_box()),spp=256)
#mi.Bitmap(img).write('cbox.exr')
#mi.image(img)
plt.axis("off")
plt.imshow(img ** (1.0/2.2))
```

**Comment by njroussel** [comment, marked as answer]
Hi @jantma2 

You can check your `PATH` envvar from python with `import os; print(os.environ['PATH'])`. This should let you check if LLVM did indeed add it to your PATH.
Alternatively, you can manually use the `DRJIT_LIBLLVM_PATH` (you can also set it in Python before importing `mitsuba`/`drjit`).One common pitfall on Windows is that this envrionment variable must point directly to the file rather than the folder which contains `LLVM-C.dll`

**Reply by jantma2** [reply]
Hi,

I solved the issue half way.
I found that I accidentally did the setup of LLVM as a common user and not as an administrator. 
There was no error but some access rights seem be wrong.

Anyway I uninstalled everything and performed the setup as an administrator and magically it works (almost).
Almost, because I've to run my script as administrator too. If I try to execute it as simple user the error with the environment variable persists. Unless I've checked that it is set correctly in "path" as also in "DRJIT_LIBLLVM_PATH".

For the beginning I can deal with the remaining limitation.

**Reply by njroussel** [reply]
That's good to hear. I'm unfortunately not familiar enough with these Windows-specific setup issues.

**Comment by jantma2** [comment]
Makes sure that administrative rights are used to install LLVM and run also for running the script. Shall this discussion being closed ?
